### PR TITLE
[TechDocs] OpenStack Swift Lowercase Migration Support Added

### PR DESCRIPTION
**Pull request by mertcb**
## Hey, I just made a Pull Request!

This PR adds support for TechDocs Beta migration for OpenStack Swift. Also this PR fixes the issue of lowercase routing, which doesn't same in every location (i.e API -> apı in Turkish) 
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hi! Would you mind rebasing this on a fresh master? I see that your yarn.lock has changes in it that were our fault and have been fixed recently.

**Comment by mertcb** [comment]
done that @freben

**Comment by mertcb** [comment]
any updates? @freben @iamEAP

File: yarn.lock
Comment (iamEAP) [inline yarn.lock:29452]: This looks like the only file that requires `@reviewers` review, and the change is not material?  So I'd recommend reverting this.

File: .changeset/chatty-months-report.md
Comment (iamEAP) [inline .changeset/chatty-months-report.md:1]: Similarly, if you rename this `techdocs-chatty-months-report.md`, we can avoid maintainer review.

File: .changeset/chatty-months-report.md
Comment (iamEAP) [inline .changeset/chatty-months-report.md:2]: While we're on the `0.x.y` series, we're treating minor API additions as patch and breaking changes as minor.  So I'd recommend setting these both to `patch` for now.

File: .changeset/chatty-months-report.md
Comment (iamEAP) [inline .changeset/chatty-months-report.md:3]: Might also break this out into its own changeset too, since the fixes to `@backstage/plugin-techdocs` are unrelated to OpenStack swift.

**Review by iamEAP — COMMENTED** [review]
Thanks for your contribution @mertcb!  Naturally, I'm unable to actually test/validate the fix, but it looks good from my POV.

Just a few house-keeping comments after which we should be ready to go.

File: yarn.lock
Comment (mertcb) [reply yarn.lock:29452]: couldn't understand that, why does it take as change, they are same I guess...

Comment (iamEAP) [reply yarn.lock:29452]: Looks like maybe fixed by [this](https://github.com/backstage/backstage/pull/8052), if you rebase?

Comment (mertcb) [reply yarn.lock]: done :)

**Review by iamEAP — APPROVED** [review]
Thanks @mertcb!  Good-to-go from my perspective.

**Comment by mertcb** [comment]
are we gonna wait for backstage/reviewers approval? @iamEAP

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
Approved! I'll let @iamEAP merge when convenient

**Pull request merged**
